src/riak_kv_pb_index.erl

```diff
@@ -133,44 +159,67 @@ maybe_perform_query({ok, Query}, Req, State) ->
                  _ -> true
              end,
     Opts0 = [{max_results, MaxResults}] ++ [{pagination_sort, PgSort} || PgSort /= undefined],
-    Opts = riak_index:add_timeout_opt(Timeout, Opts0),
+    Opts1 = riak_index:add_timeout_opt(Timeout, Opts0),
+    Opts = maybe_add_cover(Cover, Opts1),
     ReturnTerms =  riak_index:return_terms(ReturnTerms0, Query),
     QueryResult = Client:get_index(Bucket, Query, Opts),
-    handle_query_results(ReturnTerms, MaxResults, QueryResult , State).
+    handle_query_results(ReturnBody, ReturnTerms, MaxResults, QueryResult , State).
 
 
-handle_query_results(_, _, {error, Reason}, State) ->
+handle_query_results(_, _, _, {error, Reason}, State) ->
     {error, {format, Reason}, State};
-handle_query_results(ReturnTerms, MaxResults,  {ok, Results}, State) ->
+handle_query_results(ReturnBody, ReturnTerms, MaxResults,  {ok, Results}, State) ->
     Cont = make_continuation(MaxResults, Results, length(Results)),
-    Resp = encode_results(ReturnTerms, Results, Cont),
+    Resp = encode_results(response_type(ReturnBody, ReturnTerms), Results, Cont),
     {reply, Resp, State}.
 
 query_params(#rpbindexreq{index=Index= <<"$bucket">>,
                           term_regex=Re, max_results=MaxResults,
-                          continuation=Continuation}) ->
+                          continuation=Continuation, return_body=ReturnBody}) ->
     [{field, Index},
      {return_terms, false}, {term_regex, Re},
-     {max_results, MaxResults}, {continuation, Continuation}];
+     {max_results, MaxResults}, {continuation, Continuation},
+     {return_body, ReturnBody}];
 query_params(#rpbindexreq{qtype=eq, index=Index, key=Value,
                           term_regex=Re, max_results=MaxResults,
-                          continuation=Continuation}) ->
+                          continuation=Continuation, return_body=ReturnBody}) ->
     [{field, Index}, {start_term, Value}, {end_term, Value}, {term_regex, Re},
-     {max_results, MaxResults}, {return_terms, false}, {continuation, Continuation}];
+     {max_results, MaxResults}, {return_terms, false}, {continuation, Continuation},
+     {return_body, ReturnBody}];
 query_params(#rpbindexreq{index=Index, range_min=Min, range_max=Max,
                           term_regex=Re, max_results=MaxResults,
-                          continuation=Continuation}) ->
+                          continuation=Continuation, return_body=ReturnBody}) ->
      [{field, Index}, {start_term, Min}, {end_term, Max},
       {term_regex, Re}, {max_results, MaxResults},
-      {continuation, Continuation}].
+      {continuation, Continuation}, {return_body, ReturnBody}].
+
+%% Return `keys', `terms', or `objects' depending on the value of
+%% `return_body' and `return_terms'
```

Comment:
I think you ought to note that this function mirrors the functions in the riak-erlang-client code.